cni/pkg/ebpf/server/redirectionServer.go

```diff
@@ -0,0 +1,594 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+//go:generate go run github.com/cilium/ebpf/cmd/bpf2go -cc clang -target bpf -cflags "-D__TARGET_ARCH_x86"  ambient_redirect ../app/ambient_redirect.bpf.c
+//go:generate sh -c "echo '// Copyright Istio Authors' > banner.tmp"
+//go:generate sh -c "echo '//' >> banner.tmp"
+//go:generate sh -c "echo '// Licensed under the Apache License, Version 2.0 (the \"License\");' >> banner.tmp"
+//go:generate sh -c "echo '// you may not use this file except in compliance with the License.' >> banner.tmp"
+//go:generate sh -c "echo '// You may obtain a copy of the License at' >> banner.tmp"
+//go:generate sh -c "echo '//' >> banner.tmp"
+//go:generate sh -c "echo '//     http://www.apache.org/licenses/LICENSE-2.0' >> banner.tmp"
+//go:generate sh -c "echo '//' >> banner.tmp"
+//go:generate sh -c "echo '// Unless required by applicable law or agreed to in writing, software' >> banner.tmp"
+//go:generate sh -c "echo '// distributed under the License is distributed on an \"AS IS\" BASIS,' >> banner.tmp"
+//go:generate sh -c "echo '// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.' >> banner.tmp"
+//go:generate sh -c "echo '// See the License for the specific language governing permissions and' >> banner.tmp"
+//go:generate sh -c "echo '// limitations under the License.\n' >> banner.tmp"
+//go:generate sh -c "cat banner.tmp ambient_redirect_bpf.go > tmp.go && mv tmp.go ambient_redirect_bpf.go && rm banner.tmp"
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/netip"
+	"os"
+
+	"github.com/cilium/ebpf"
+	"github.com/containernetworking/plugins/pkg/ns"
```

Comment:
Need to verify, but I think may have broken building on the Mac. Will need a Linux and non_linux version of this file.

Reply:
No, I think this was me in https://github.com/istio/istio/pull/43578

See https://github.com/istio/istio/issues/43689#issuecomment-1450401499

Reply:
Well, I did test `make build` before this PR and it worked, and after this PR it does not, so certainly this PR caused part of the issue. Certainly, another PR could have caused a bigger issue.